packages/medusa/src/types/product-variant.ts

```diff
@@ -62,7 +62,7 @@ export type CreateProductVariantInput = {
 
 export type UpdateProductVariantInput = {
   title?: string
-  product_id: string
+  product_id?: string
```

Comment:
Having the `product_id` as non optional causes an error [here](https://github.com/medusajs/medusa/blob/f1b06958c2108ce98f5e8a25e6427b57ffe38ef2/packages/medusa/src/services/product.ts#L667-L672), of course one easy fix is to simply do this:
```typescript
const saved = await this.productVariantService_
  .withTransaction(manager)
  .update(variant, {
    ...newVariant,
    variant_rank,
    product_id: variant.product_id
  })
```
But I thought since we're using typeorm's `save` method [here](https://github.com/medusajs/medusa/blob/develop/packages/medusa/src/services/product-variant.ts#L369), which supports partial updates, the `product_id` doesn't have to be defined. Wdyt?

Reply:
fine for me, I just checked and we can update any variants, what you've done is nice :) the product service already check that the variants belongs to the product 👍